packages/flutter_tools/lib/src/macos/cocoapod_utils.dart

```diff
@@ -37,7 +35,7 @@ Future<void> processPodsIfNeeded(
 
   final bool didPodInstall = await globals.cocoaPods.processPods(
     xcodeProject: xcodeProject,
-    engineDir: flutterFrameworkDir(buildMode),
```

Comment:
This is wrong on macOS and is pointing to the iOS engine artifacts directory, not sure how (if) it's working on macOS...